src/app/modules/submission/multirespondent-submission/multirespondent-submission.middleware.ts

```diff
@@ -446,18 +451,40 @@ export const validateMultirespondentSubmission = async (
                   previousSubmissionDecryptedContent.responses as FieldResponsesV3
 
                 return Result.combine(
-                  nonEditableFieldIdsWithResponses.map((fieldId) =>
-                    isFieldResponseV3Equal(
-                      req.body.responses[fieldId],
-                      previousResponses[fieldId],
+                  nonEditableFieldIdsWithResponses.map((fieldId) => {
+                    const incomingResField = req.body.responses[fieldId]
+                    const prevResField = previousResponses[fieldId]
+
+                    const resp = isFieldResponseV3Equal(
+                      incomingResField,
+                      prevResField,
                     )
-                      ? ok(undefined)
-                      : err(
-                          new ProcessingError(
-                            'Submitted response on a non-editable field which did not match previous response',
-                          ),
+
+                    if (!resp) {
+                      return err(
+                        new ProcessingError(
+                          'Submitted response on a non-editable field which did not match previous response',
                         ),
-                  ),
+                      )
+                    }
+
+                    /**
+                     * Files are verified to have the same md5 hash, so we can safely assume that the files are the same.
+                     * We should also ignore attachment names from submissions as handleDuplicatesInAttachments may rename files
+                     */
+                    if (
+                      incomingResField.fieldType === BasicField.Attachment &&
+                      prevResField.fieldType === BasicField.Attachment
+                    ) {
+                      incomingResField.answer.answer =
+                        prevResField.answer.answer
+
+                      incomingResField.answer.filename =
+                        prevResField.answer.answer
+                    }
+
+                    return ok(undefined)
```

Comment:
what's the point of this? this replaces the response body data with the original filenames right. was this the case where someone downloads an attachment, renames it and re-uploads the same attachment?

Reply:
This is to handle the case where the attachment name change that is caused by [`handleDuplicatesInAttachments`](https://github.com/opengovsg/FormSG/blob/develop/src/app/modules/submission/receiver/receiver.utils.ts#L133)

Reply:
gotcha. attachments are so annoying

so let's say R1 submits attachment Q1: `a.txt`, and R2 submits attachment Q2: `a.txt` but the two files are different.  `handleDuplicatesInAttachments` will rename Q1 to `1-a.txt` in `q1.answer.answer`, but this code snippet restores the `q1.answer.answer` to be `a.txt` - am i understanding this correctly?

Reply:
Yes exactly.

Reply:
gotcha. thanks!